packages/browser/src/utils/survey-utils.ts

```diff
@@ -39,14 +42,90 @@ export const getSurveySeenKey = (survey: Pick<Survey, 'id' | 'current_iteration'
     return surveySeenKey
 }
 
-export const setSurveySeenOnLocalStorage = (survey: Pick<Survey, 'id' | 'current_iteration'>) => {
-    const isSurveySeen = localStorage.getItem(getSurveySeenKey(survey))
-    // if survey is already seen, no need to set it again
+export function isPersistenceEnabledWithLocalStorage(
+    posthog?: PostHog
+): posthog is PostHog & { persistence: PostHogPersistence } {
+    return !!(
+        posthog?.persistence &&
+        !posthog.persistence.isDisabled() &&
+        posthog.config?.persistence?.includes('localStorage')
+    )
+}
+
+/**
+ * We used to retrieve from localStorage directly. Now, we instead rely on the persistence API, since this is the preferred way to store data.
+ * But, since we might have customers that might be using surveys with persistence disabled, we need to fallback to localStorage
+ * to maintain backwards compatibility.
```

Comment:
i think the `persistence` config should mention that surveys will fallback to localStorage if needed

Reply:
just adde